### PR TITLE
Icu 13302 add filtering pagination support for scopes

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-13302

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-13302)

## Description

Add hds::table to Orgs and Projects.
Add search & pagination support.
Refactor usage of `this.store.query('scopes' ...` in all necessary places.

## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/107949262/83a9c025-1053-4f8e-8a09-09b6cf58fdbb

## How to Test

1. start boundary dev env
2. Clear any existing indexedDB from applications tab in dev tools
3. Navigate to Orgs & Projects tab
4. Validate that search works as expected for both list views
5. Validate CRUDL for org & project scopes remains functional

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [X] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [X] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/addon/services/indexed-db.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ export const modelIndexes = {
     '&id, attributes.created_time, attributes.type, attributes.status, attributes.endpoint, attributes.target_id, attributes.user_id, attributes.scope.scope_id',
   user: '&id, attributes.created_time, attributes.name, attributes.description, attributes.scope.scope_id',
   role: '&id, attributes.created_time, attributes.name, attributes.description, attributes.scope.scope_id',
+  scope:
+    '&id, attributes.created_time, attributes.type, attributes.name, attributes.description, attributes.scope.scope_id',
```
Comment (lisbet-alvarez) [inline addons/api/addon/services/indexed-db.js:18]: I added 'attributes.type` since it could be useful

**Review by cameronperera — APPROVED** [review]

File: ui/admin/app/routes/scopes/scope/scopes.js [hunk 2]
```diff
@@ -26,23 +25,4 @@ export default class ScopesScopeScopesRoute extends Route {
...
-    const currentScope = this.modelFor('scopes.scope');
-    const parentScope = !currentScope.isGlobal
-      ? await this.store.findRecord('scope', currentScope.scopeID)
-      : null;
-    const subScopes = await this.store.query('scope', {
-      scope_id: currentScope.id,
-    });
-    return {
-      currentScope,
-      parentScope,
```
Comment (ZedLi) [inline ui/admin/app/routes/scopes/scope/scopes.js:44]: Was `parentScope` just never used?

**Review by ZedLi — APPROVED** [review]
Great job! Looks nice!

File: ui/admin/app/routes/scopes/scope/scopes.js (hunk 2 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/scopes.js:44]: yup, it wasn't being used.

File: ui/admin/tests/acceptance/targets/brokered-credential-sources-test.js [hunk 3]
```diff
@@ -246,7 +246,7 @@ module('Acceptance | targets | brokered credential sources', function (hooks) {
         (item) => item !== 'add-credential-sources',
       );
     await visit(urls.brokeredCredentialSources);
-    assert.dom('.rose-dropdown-content a:nth-child(2)').doesNotExist();
+    assert.dom('[data-test-add-brokered-cred-sources-action]').doesNotExist();
```
Comment (lisbet-alvarez) [inline ui/admin/tests/acceptance/targets/brokered-credential-sources-test.js:249]: this selector was incorrect, so updated to use `data-test-*`

File: ui/admin/app/routes/onboarding.js [hunk 4]
```diff
@@ -25,6 +26,8 @@ export default class OnboardingRoute extends Route {
   async beforeModel() {
     const orgs = await this.store.query('scope', {
       scope_id: 'global',
+      recursive: true,
```
Comment (ZedLi) [inline ui/admin/app/routes/onboarding.js:29]: One last small thing 🙏 , can we actually make the requests match what they were previously, i.e. if they were using global scope and recursive, send both but if they were just using global scope but not recursive, don't include the recursive query param? 

As part of the changes I'm making to how we query, the params matter and the query filter below doesn't recursively return all the data anyways. We don't have an easy way to return data recursively without doing the filters manually right now

**Review by ZedLi — APPROVED** [review]
Looks great!

**Review by cameronperera — APPROVED** [review]
Great work!

File: ui/admin/app/routes/onboarding.js (hunk 4 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/onboarding.js]: Of course! Updated!

**Pull request merged**